Short-Answer/Algorithms_Answers.md

```diff
@@ -3,13 +3,26 @@
 ## Exercise I
 
 a)
+O(n), linear.
+This is linear because the while loops end up running n times before meeting the condition
 
 
 b)
+O(n), linear.
```

Comment:
We multiply the individual runtimes of each loop to obtain the runtime of the entire block since these loops are nested, giving us an overall runtime of O(n log(n)).
```
sum = 0
    for i in range(n):
      j = 1
      while j < n:
        j *= 2
        sum += 1
```